lsp/server/src/diagnostic/gql/over-sized-record.ts

```diff
@@ -0,0 +1,167 @@
+    }
+
+    getId(): string {
+        return RULE_ID;
+    }
+}
+
+export interface OverSizedDiagnostics {
+    overSizedFields: Array<FieldNode>;
+    overSizedEntities: Array<FieldNode>;
+}
+export async function createDiagnostics(
+    rootNode: RootNode
+): Promise<OverSizedDiagnostics> {
+    const results: OverSizedDiagnostics = {
+        overSizedFields: [],
+        overSizedEntities: []
+    };
+
+    for (const operationNode of rootNode.operations) {
+        for (const entityNode of operationNode.entities) {
+            try {
+                await generateDiagnostic(entityNode, results);
+            } catch (e) {
+                console.log(
+                    `Cannot conduct over-sized record diagnostic for ${entityNode.name} `
+                );
+            }
+        }
+    }
+    return results;
+}
+
+/**
+ * Recursively search within an EntityNode for FieldNodes containing oversized records.
+ * @param entityNode The EntityNode to search within
+ * @param overSizedFields Array to which oversized FieldNodes are added.
+ * @returns A promise that resolves when diagnostics generation is complete.
+ */
+async function generateDiagnostic(
+    entityNode: EntityNode,
+    overSizedDiagnostic: OverSizedDiagnostics
+) {
+    if (entityNode.name) {
+        const objectInfo = await OrgUtils.getObjectInfo(entityNode.name);
+        if (objectInfo === undefined) {
+            return;
+        }
+
+        let totalSize = 0;
+        for (const propertyNode of entityNode.properties) {
+            const fieldSize = getFieldSize(objectInfo, propertyNode.property);
+
+            propertyNode.size = fieldSize;
+
+            if (fieldSize !== undefined) {
+                totalSize += fieldSize;
+
+                // Oversized field check
+                if (fieldSize > MAX_ALLOWED_SIZE) {
+                    overSizedDiagnostic.overSizedFields.push(propertyNode.node);
+                }
+            }
+        }
+        entityNode.size = totalSize;
+        if (totalSize > MAX_ALLOWED_SIZE) {
+            overSizedDiagnostic.overSizedEntities.push(entityNode.node);
+        }
+
+        for (const relation of entityNode.relationships) {
+            if (Array.isArray(relation.entity)) {
+                continue;
+            }
+            // Find entity name. Polymorphic parent relationship is not supported for now.
+            if (relation.entity.name === undefined) {
+                const entityName = resolveEntityNameFromMetadata(
+                    objectInfo,
+                    relation
+                );
+                if (entityName === undefined) {
+                    continue;
+                }
+                relation.entity.name = entityName;
+            }
+
+            try {
+                await generateDiagnostic(relation.entity, overSizedDiagnostic);
+            } catch (e) {
+                console.log(
+                    `Cannot conduct over-sized record diagnostic for ${entityNode.name} `
```

Comment:
Same here, let's log error's message since we're logging the entity with problem already.

Reply:
added